pkg/helpers/contexts/heartbeat.go

```diff
@@ -56,5 +56,6 @@ func (ctx *heartBeatCtx) heartBeat() {
 func (ctx *heartBeatCtx) heartBeatStop() {
 	ctx.mutex.Lock()
 	defer ctx.mutex.Unlock()
+	defer ctx.Cancel()
```

Comment:
is there any reason why we want to defer this? ctx.Cancel()?

Reply:
Not because a specific reason. Just because I prefer to cancel the context after stopping the timer and as timer.Stop() can panic (it shouldn't), assure its execution.